ui/src/components/search.js

```diff
@@ -0,0 +1,102 @@
+import ItemElement from './itemELement';
+import { Link } from 'react-router-dom';
+import {getAllCategories} from '../apiConsumer/categoryConsumer';
+import {searchItems} from '../apiConsumer/itemFetchConsumer' 
+
+class Search extends React.Component{
+
+    constructor(props){
+        super(props);
+        this.loadCount=0;
+        this.searchText="";
+        this.selectedCategories=[];
+        this.state={
+            items:[],
+            loadMore: true,
+            categories:[],
+            selectedCategories:[]
+        }
+    }
+        
+    componentDidMount=()=>{
+        this.loadCategories();
+        this.load();
+        this.context.setSearchCallback((text)=>{this.textChanged(text);});
+    }
+
+    textChanged=(text)=>{
+        this.setState({['loadMore']:true});
+        this.loadCount=0;
+        this.searchText=text;
+        this.load()
+    }
+
+    load=()=>{
+        searchItems(this.searchText,this.selectedCategories,this.loadCount,12,(success, data)=>{
+            if(success){
+                if(data.length==0){
+                    this.setState({['loadMore']:false});
+                }
+                if(this.loadCount!=0){
+                    this.setState({['items']:[...this.state.items,...data]});
+                }else{
+                    this.setState({['items']:data});
+                }
+                this.loadCount=this.loadCount+1;
+            }
+        })
+    }
+
+    loadCategories=()=>{
+        getAllCategories((success,data)=>{
+            if(success){
+                this.setState({['categories']: data.slice(1)})
+            }else{
+                console.log("Cannot fetch categories")
+            }
+        });
+    }
+
+    selectCategory=(cat)=>{
+        if(cat==""){
+            this.setState({['selectedCategories']:[]});
+            this.selectedCategories=[];
+        }else{
+            if(this.state.selectedCategories.includes(cat)){
+                this.setState({['selectedCategories']:this.state.selectedCategories.filter((x)=>x!=cat)});
+                this.selectedCategories=this.selectedCategories.filter((x)=>x!=cat);
+            }else{
+                this.setState({['selectedCategories']:[...this.state.selectedCategories,cat]})
+                this.selectedCategories.push(cat);
+            }
+        }
+        this.loadCount=0;
+        this.setState({['loadMore']:true});
+        this.load()
+    }
+
+    render(){
+        return(
+        <div class="shopContainer">
+           
+            <div class="categoriesList">
+                <div class={this.state.selectedCategories.length==0?"categoryButtonSelected":"categoryButton"} onClick={()=>this.selectCategory("")}>all categories</div>
+                {this.state.categories.map(category=><div class={this.state.selectedCategories.includes(category.id)?"categoryButtonSelected":"categoryButton"} onClick={()=>this.selectCategory(category.id)}>{category.name}</div>)}
```

Comment:
break these 2 long lines into multiple

also, think you are getting errors also, but we should use className instead of class in react (because class i reserved keyword)